ctest-next/src/generator.rs

```diff
@@ -76,9 +99,183 @@ impl TestGenerator {
+            }
+        }));
+        self
+    }
+
+    /// Configures whether the tests for a static definition are generated.
+    pub fn skip_static(&mut self, f: impl Fn(&Static) -> bool + 'static) -> &mut Self {
+        self.skips.push(Box::new(move |item| {
+            if let MapInput::Static(static_) = item {
+                f(static_)
+            } else {
+                false
+            }
+        }));
+        self
+    }
+
+    /// Configures whether tests for a function definition are generated.
+    pub fn skip_fn(&mut self, f: impl Fn(&crate::Fn) -> bool + 'static) -> &mut Self {
+        self.skips.push(Box::new(move |item| {
+            if let MapInput::Fn(func) = item {
+                f(func)
+            } else {
+                false
+            }
+        }));
+        self
+    }
+
+    /// Sets the programming language.
+    pub fn language(&mut self, language: Language) -> &mut Self {
+        self.language = language;
+        self
+    }
+
+    /// Add a flag to the C compiler invocation.
+    pub fn flag(&mut self, flag: &str) -> &mut Self {
+        self.flags.push(flag.to_string());
+        self
+    }
+
+    /// Set a `-D` flag for the C compiler being called.
+    ///
+    /// This can be used to define various variables to configure how header
+    /// files are included or what APIs are exposed from header files.
+    pub fn define(&mut self, k: &str, v: Option<&str>) -> &mut Self {
+        self.defines
+            .push((k.to_string(), v.map(std::string::ToString::to_string)));
+        self
+    }
+
+    /// Configures how Rust `const`s names are translated to C.
+    pub fn map_constant(&mut self, f: impl Fn(&Const) -> Option<String> + 'static) -> &mut Self {
+        self.mapped_names.push(Box::new(move |item| {
+            if let MapInput::Const(c) = item {
+                f(c)
+            } else {
+                None
+            }
+        }));
+        self
+    }
+
+    /// Configures how a Rust struct field is translated to a C struct field.
+    pub fn map_field(
+        &mut self,
+        f: impl Fn(&Struct, &Field) -> Option<String> + 'static,
+    ) -> &mut Self {
+        self.mapped_names.push(Box::new(move |item| {
+            if let MapInput::Field(s, c) = item {
+                f(s, c)
+            } else {
+                None
+            }
+        }));
+        self
+    }
+
+    /// Configures the name of a function in the generated C code.
+    pub fn map_fn(&mut self, f: impl Fn(&crate::Fn) -> Option<String> + 'static) -> &mut Self {
+        self.mapped_names.push(Box::new(move |item| {
+            if let MapInput::Fn(func) = item {
+                f(func)
+            } else {
+                None
+            }
+        }));
+        self
+    }
+
+    /// Configures how a Rust type is translated to a C type.
+    pub fn map_type(
+        &mut self,
+        f: impl Fn(&str, bool, bool) -> Option<String> + 'static,
+    ) -> &mut Self {
```

Comment:
Maybe `rename_fn` and `rename_type`? Also, what are the `bool`s in `map_type`?

Reply:
Ah, looks like `is_struct` and `is_union` from `ctest`. Maybe an `#[non_exhaustive] enum TyKind { Struct, Union }` instead?